code/modules/library/lib_items.dm

```diff
@@ -61,7 +61,7 @@
 				state = 0
 
 		if(2)
-			if(istype(I, /obj/item/weapon/book) || istype(I, /obj/item/weapon/spellbook))
+			if(istype(I, /obj/item/weapon/book) || istype(I, /obj/item/weapon/spellbook) || istype(I, /obj/item/weapon/storage/book))

```

Comment:
it's sad that this is how this has to be done, However:

replace those 3 (4 with tk's) typepaths with a list containing those typepaths, and just run is_type_in_list() on the list. call it allowed_books or something.

```
var/list/allowed_books = list(/obj/item/weapon/book, /obj/item/weapon/spellbook, etc.)
if(is_type_in_list(I, allowed_books))
    do_stuff()
```